web-admin/src/features/organizations/users/AddUsersDialog.svelte

```diff
@@ -98,37 +94,53 @@
       SPA: true,
       validators: schema,
       async onUpdate({ form }) {
+        // Reset failed invites
+        failedInvites = [];
+
         if (!form.valid) return;
         const values = form.data;
         const emails = values.emails.map((e) => e.trim()).filter(Boolean);
         if (emails.length === 0) return;
 
         const succeeded = [];
-        let errored = false;
-        await Promise.all(
-          emails.map(async (email) => {
-            try {
-              await handleCreate(email, values.role, isSuperUser);
-              succeeded.push(email);
-            } catch (error) {
-              console.error("Error adding user to organization", error);
-              errored = true;
-            }
-          }),
-        );
-
-        eventBus.emit("notification", {
-          type: "success",
-          message: `Invited ${succeeded.length} ${succeeded.length === 1 ? "person" : "people"} as ${values.role}`,
-        });
-
-        if (errored) {
+        const failed = [];
+
+        // Process invites sequentially to maintain order
+        for (const email of emails) {
```

Comment:
Hmmm not sure doing this serially is a good idea. There could a huge list of invitees leading to a slow experience.

How about keeping the `Promise.all` but use the index in the `emails.map` to maintain order?
```
// pre-allocate
succeeded = new Array(emails.length).fill(undefined)
failed = ...

await Promise.all(
  emails.map(async (email, i) => {
    try {
      ...
      succeeded[i] = email
    } catch (err) {
      failed[i] = err
    }
  })
)

// remove empty values
succeeded = succeeded.filter(Boolean)
failed = failed.filter(Boolean)
```